### PR TITLE
feat: [290083] Scope changes on routes

### PR DESCRIPTION
**Pull request by KiranVarma02**
https://dev.azure.com/defragovuk/DEFRA-CDP/_boards/board/t/DXT%20Team/Stories?System.IterationPath=%40currentIteration&workitem=290083

### CODE REVIEW

File: designer/server/src/common/helpers/auth/user-session.js [hunk 1]
```diff
@@ -64,6 +65,18 @@ export async function createUserSession(request, artifacts) {
...
+    auth: {
+      mode: 'try',
+      access: {
+        entity: 'user',
+        scope: [`+${scope}`]
+      }
+    }
+  }
+}
+
```
Comment (KiranVarma02) [inline designer/server/src/common/helpers/auth/user-session.js:79]: I didn't find any other way to set options in 1 place for all routes. There is probably a better way to do this ?!

File: designer/server/src/routes/forms/create.js [hunk 2]
```diff
@@ -187,6 +201,15 @@ export default [
...
       )
+    },
+    options: {
+      auth: {
+        mode: 'required',
+        access: {
+          entity: 'user',
+          scope: [`+${scopes.SCOPE_READ}`]
+        }
+      }
```
Comment (KiranVarma02) [inline designer/server/src/routes/forms/create.js:212]: this doesn't have lint issue but don't want to repeat this for all routes

File: designer/server/src/routes/account/auth.js [hunk 3]
```diff
@@ -26,24 +24,15 @@ export default [
...
 
-      /**
-       * Drop user sessions when scopes are missing but let users sign back in
-       * @todo Keep users signed in but restrict route access by scope
-       */
-      if (!credentials.scope.includes(scopes.SCOPE_WRITE)) {
-        await dropUserSession(request)
-
-        // Redirect users to a help message
-        yar.flash(sessionNames.userAuthFailed, true)
```
Comment (alexluckett) [inline designer/server/src/routes/account/auth.js:37]: Note to other reviewers, we will pick this back up in the next PR.

**Review by alexluckett — COMMENTED** [review]
Can you add scope write for `/library/{slug}/editor`, all in `form-lifecycle.js` and `/api/{id}/data` too, please?

**Comment by KiranVarma02** [comment]
> Can you add SCOPE_WRITE for `/library/{slug}/editor`, all in `form-lifecycle.js` and `/api/{id}/data` too, please?

`form-lifecycle.js` was already done and for others I just pushed changes 👍

File: designer/server/src/routes/account/auth.test.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ jest.mock('~/src/lib/forms.js')
 describe('Authentiation', () => {
   /** @type {import('@hapi/hapi').Server} */
   let server
+  const testingScopes = [auth, authScopesEmpty, authGroupsInvalid]
```
Comment (colinrotherham) [inline designer/server/src/routes/account/auth.test.js:13]: If we're only using this for a single `describe.each()` can we put it inside?

```js
  describe.each(
    auth,
    authScopesEmpty,
    authGroupsInvalid
  )('With valid or invalid scopes', (authIn) => {
```

File: designer/server/src/routes/account/auth.js (hunk 3 above)
Comment (alexluckett) [reply designer/server/src/routes/account/auth.js:37]: Resolved in https://github.com/DEFRA/forms-designer/pull/385/commits/8e801b47a3b3fbe31e064af0d74283eaf3f3b1a7

File: designer/server/src/routes/account/auth.test.js (hunk 4 above)
Comment (KiranVarma02) [reply designer/server/src/routes/account/auth.test.js:13]: I am getting type errors, is it worth adding types in test? or shall i leave it as it is now?

Comment (colinrotherham) [reply designer/server/src/routes/account/auth.test.js:13]: Sorry I missed the wrapping `[]` in my suggestion above (fixed)

Comment (KiranVarma02) [reply designer/server/src/routes/account/auth.test.js:13]: ofcourse :), I was not paying enough attention to fix that myself :)

File: designer/server/src/createServer.ts [hunk 5]
```diff
@@ -5,6 +5,8 @@ import inert from '@hapi/inert'
 import Wreck from '@hapi/wreck'
 import { ProxyAgent } from 'proxy-agent'
 
+import { SCOPE_READ } from './common/constants/scopes.js'
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply designer/server/src/createServer.ts]: done

File: designer/server/src/routes/account/auth.test.js (hunk 4 above)
Comment (KiranVarma02) [reply designer/server/src/routes/account/auth.test.js]: done

File: designer/server/src/createServer.ts [hunk 6]
```diff
@@ -5,6 +5,9 @@ import inert from '@hapi/inert'
 import Wreck from '@hapi/wreck'
 import { ProxyAgent } from 'proxy-agent'
 
+import errorPage from './plugins/errorPage.js'
```
Comment (colinrotherham) [inline designer/server/src/createServer.ts:8]:
```suggestion
import errorPage from '~/src/plugins/errorPage.js'
```

File: designer/server/src/plugins/errorPage.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import {
+  type ServerRegisterPluginObject,
+  type Request,
+  type ResponseToolkit
+} from '@hapi/hapi'
+
+import { unauthorisedViewModel } from '../models/errors.js'
```
Comment (colinrotherham) [inline designer/server/src/plugins/errorPage.ts:7]:
```suggestion
import { unauthorisedViewModel } from '~/src/models/errors.js'
```

**Review by colinrotherham — APPROVED** [review]
Added a couple of comments

Approving for further testing, I've not checked the scopes fully but I'm sure we'll find any issues if there are any. Thanks @KiranVarma02

**Pull request merged**